### PR TITLE
Navigation User Stories

### LINKED ISSUE #1: User Story 2, Visitor Navigation
As a visitor
I see a navigation bar
This navigation bar includes links for the following:
- a link to return to the welcome / home page of the application ("/")
- a link to browse all items for sale ("/items")
- a link to see all merchants ("/merchants")
- a link to my shopping cart ("/cart")
- a link to log in ("/login")
- a link to the user registration page ("/register")

Next to the shopping cart link I see a count of the items in my cart

### LINKED ISSUE #2: User Story 3, User Navigation
As a default user
I see the same links as a visitor
Plus the following links
- a link to my profile page ("/profile")
- a link to log out ("/logout")

Minus the following links
- I do not see a link to log in or register

I also see text that says "Logged in as Mike Dao" (or whatever my name is)

### LINKED ISSUE #3: User Story 4, Merchant Navigation
As a merchant employee
I see the same links as a regular user
Plus the following links:
- a link to my merchant dashboard ("/merchant")

### LINKED ISSUE #4: User Story 5, Admin Navigation
As an admin
I see the same links as a regular user
Plus the following links
- a link to my admin dashboard ("/admin")
- a link to see all users ("/admin/users")

Minus the following links/info
- a link to my shopping cart ("/cart") or count of cart items

### LINKED ISSUE #5: User Story 6, Visitor Navigation Restrictions
As a visitor
When I try to access any path that begins with the following, then I see a 404 error:
- '/merchant'
- '/admin'
- '/profile'

### LINKED ISSUE #6: User Story 7, User Navigation Restrictions
As a default user
When I try to access any path that begins with the following, then I see a 404 error:
- '/merchant'
- '/admin'

### LINKED ISSUE #7: User Story 8, Merchant Navigation Restrictions
As a merchant employee
When I try to access any path that begins with the following, then I see a 404 error:
- '/admin'

### LINKED ISSUE #8: User Story 9, Admin Navigation Restrictions
As an admin
When I try to access any path that begins with the following, then I see a 404 error:
- '/merchant'
- '/cart'

### PR DESCRIPTION
**Pull request by fieldstyler**
User Stories 2 - 9

Visitor Navigation and navigation restrictions
User Navigation and navigation restrictions
Merchant Navigation and navigation restrictions
Admin Navigation and navigation restrictions

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -6,7 +6,7 @@ def create
     if user = User.find_by(email: params[:email])
       if user.authenticate(params[:password])
         session[:user_id] = user.id
-        flash[:success] = "Successfully logged in"
+        flash[:success] = "Logged in as #{user.name}"
```
Comment (dcoleman21) [inline app/controllers/sessions_controller.rb:9]: This is great. Makes it more dynamic

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -10,14 +10,27 @@
 
   <body>
     <nav class = "topnav">
+      <%= link_to "Home Page", "/" %>
       <%= link_to "All Merchants", "/merchants"%>
       <%= link_to "All Items", "/items"%>
-      <%= link_to "Cart: #{cart.total_items}", "/cart" %>
-      <%= link_to "Register", '/register' %>
-      <% if current_user %>
+      <% if current_user && current_user.merchant? %>
```
Comment (dcoleman21) [inline app/views/layouts/application.html.erb:16]: Let's clarify if there is a better way to call this (per our method in the application controller).

File: spec/features/navigation/admin_spec.rb [hunk 3]
```diff
@@ -0,0 +1,72 @@
+require 'rails_helper'
+
+RSpec.describe 'admin navigation' do
+
+  describe 'admin nav bar approved pathways' do
+    it 'includes links: return to welcome page, browse all items, see all merchants, my shopping cart, log in, user registration page, and profile page' do
+      @admin = User.create!(name: 'Chilly Billy',
```
Comment (dcoleman21) [inline spec/features/navigation/admin_spec.rb:7]: Maybe we create a before each do here when refactoring

**Review by dcoleman21 — APPROVED** [review]

File: app/views/layouts/application.html.erb (hunk 2 above)
Comment (fieldstyler) [reply app/views/layouts/application.html.erb:16]: Yes, let's ask during project check-in. Great reminder!

**Pull request merged**